doc/rtd/reference/datasources/nocloud.rst

```diff
@@ -4,97 +4,237 @@ NoCloud
-    source of configuration data by the NoCloud datasource. While it may
-    be acted upon by cloud-init, using
-    :ref:`DataSourceNone<datasource_none_example>` should be preferred.
+Cloud-init discovers four types of configuration at runtime. The source of
+these configuration types is configurable with a meta-configuration. This
+meta-configuration can be delivered to cloud-init in different ways, but is
+different from the configurations that cloud-init uses to configure the
+instance at runtime.
+
+user-data
+---------
+
+User-data is a :ref:`configuration format<user_data_formats>` that allows a
+user to configure an instance.
+
+meta data
+---------
+
+The ``meta-data`` file is a YAML-formatted file. This is a required
+configuration type.
+
+vendor data (optional)
+----------------------
+
+Vendor data may be used to provide default cloud-specific configurations which
+may be overriden by user-data. This may be useful, for example, to configure an
+instance with a cloud provider's repository mirror for faster package
+installation.
+
+network config (optional)
+-------------------------
+
+Network configuration typically comes from the cloud provider to set
+cloud-specific network configurations, or a reasonable default is set by
+cloud-init (typically cloud-init brings up an interface using DHCP).
 
-Method 1: Labeled filesystem
+Since NoCloud is a generic datasource, network configuration may be set the
+same way as user-data, meta data, vendor data.
+
+See the :ref:`network configuration<network_config>` documentation for
+information on network configuration formats.
+
+Meta-configuration
+==================
+
+The purpose of the meta-configuration is to tell cloud-init where it can find
+the runtime configurations described above.
+
+There are two methods for cloud-init to receive a meta-configuration.
+
+Method 1: Line configuration
 ----------------------------
 
-A labeled `vfat`_ or `iso9660` filesystem may be used. The filesystem volume
-must be labelled ``CIDATA``.
+The "line configuration" is a single string of text which is passed to an
+instance at boot time via either the kernel command line or in the serial
+number exposed via DMI (sometimes called SMBIOS).
 
+Example: ::
 
-Method 2: Custom webserver
---------------------------
+  ds=nocloud;s=https://10.42.42.42/configs/
 
-Configuration files can be provided to cloud-init over HTTP(s). To tell
-cloud-init the URI to use, arguments must be passed to the instance via the
-kernel command line or SMBIOS serial number. This argument might look like: ::
+In the above line configuration, ``ds=nocloud`` tells cloud-init to use the
+NoCloud datasource, and ``s=https://10.42.42.42/configs/`` tells cloud-init to
+fetch configurations using ``https`` from the URI
+``https://10.42.42.42/configs/``.
 
-  ds=nocloud;s=https://10.42.42.42/cloud-init/configs/
+We will describe the possible values in a line configuration in the following
```

Comment:
Could we explicitly link to reference/datasources/nocloud.html#line-configuration-in-detail ? In that way, we are directly pointing to the user where those documentation bits are and it is hopefully easier for them.